File: internal/dict/dict.go
// Package dict は練習に使う日本語単語の読みを頻度順で提供する。
// データは SudachiDict から tools/mkdict で生成した words.txt を埋め込む。
package dict

import (
	_ "embed"
	"strings"
	"sync"
)

//go:embed words.txt
var raw string

var load = sync.OnceValue(func() []string {
	lines := strings.Split(raw, "\n")
	words := make([]string, 0, len(lines))
	for _, line := range lines {
		line = strings.TrimSpace(line)
		if line == "" || strings.HasPrefix(line, "#") {
			continue
		}
		words = append(words, line)
	}
	return words
})

// Words はひらがな読みの単語を頻度の高い順で返す。
// 返り値を書き換えてはいけない。
func Words() []string { return load() }

File: web/src/lib/dict.ts
// 練習に使う単語辞書。Go版と同じ internal/dict/words.txt を読み込む。
import raw from "@dict/words.txt?raw";

let cached: string[] | null = null;

export function words(): string[] {
  if (cached === null) {
    cached = raw
      .split("\n")
      .map((line) => line.trim())
      .filter((line) => line !== "" && !line.startsWith("#"));
  }
  return cached;
}
